test/utils

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env bash
+
+set -u
+
+cd "$( dirname "${BASH_SOURCE[0]}" )" || return
+
+function teardown() {
+    EXIT_CODE="$(tr -d 0 < "$OUTPUT_DIR/EXIT_CODE.txt")"
+    rm -r "$INPUT_DIR" "$OUTPUT_DIR"
+    exit "${EXIT_CODE:-0}"
+}
+
+INPUT_DIR="$(mktemp -d)"
+OUTPUT_DIR="$(mktemp -d)"
+trap "teardown" EXIT
+
+function run_test() {
+
+    test_name="$1"
+    command_name="$2"
+    shift 2
+
+    echo "⋯ $command_name $test_name"
```

Comment:
Not sure whether to be happy about seeing unicode in shell scripts.

Reply:
I wasn't too happy about that either. Okay, I took the time to implement full TAP output according to the specification now... with the exception of running counts.